pilot/pkg/serviceregistry/kube/controller_test.go

```diff
@@ -561,15 +561,22 @@ func TestController_Service(t *testing.T) {
 	if len(svcList) != len(expectedSvcList) {
 		t.Errorf("Expecting %d service but got %d\r\n", len(expectedSvcList), len(svcList))
 	}
-	for i, exp := range expectedSvcList {
-		if exp.Hostname != svcList[i].Hostname {
-			t.Errorf("got hostname of %dst service, got:\n%#v\nwanted:\n%#v\n", i, svcList[i].Hostname, exp.Hostname)
-		}
-		if exp.Address != svcList[i].Address {
-			t.Errorf("got address of %dst service, got:\n%#v\nwanted:\n%#v\n", i, svcList[i].Address, exp.Address)
+	for _, exp := range expectedSvcList {
```

Comment:
We can define `expectedSvcList` as a map like `expectedSvcList := map[model.Hostname]*model.Service{}`, then the expected result can be directly got by hostname.

Reply:
I prefer `Controller.Services ` not having a random output.

Reply:
Thanks @supereagle . It will make the code a little more cleaner so I will take your suggestion.

Reply:
> I prefer `Controller.Services` not having a random output.

I have on opinion about the sorted output of `Controller.Services`, but I think test cases should not depend on the order of the output, otherwise it will be very easy to make mistakes for developers.